### PR TITLE
Fix `createContractWithStakingFields()`

### LINKED ISSUE #8719: Implement staking id validation logic for `ContractCreate` transactions
We have the `createContractWithStakingFields` test from `ContractCreateSuite` failing due to this missing.
I think we should implement the equivalent logic from `ContractCreateTransitionLogic.validate()` in mono-service, in particular [this](https://github.com/hashgraph/hedera-services/blob/c672967964359c08a8b21c4a17563a1b3e599f51/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCreateTransitionLogic.java#L335).

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #8719 
   * The remaining issue with the `createContractWithStakingFields()` spec was that a `_TO_ACCOUNT` stake change scenario would cause a `ConcurrentModificationException` in the `for` loop [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java#L105), since the logic _inside_ the loop would add the "stakee" account to the modified set---and cause the iterator to detect a concurrent modification.
- Adds a spec validating system accounts cannot be used as initcode.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
Makes sense to me!

**Review by Nana-EC — APPROVED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java [hunk 1]
```diff
@@ -100,7 +101,10 @@ public Map<AccountID, Long> applyStakingRewards(final FinalizeContext context) {
      * @param writableStore The store to write to for updated values
      */
     public void adjustStakedToMeForAccountStakees(@NonNull final WritableAccountStore writableStore) {
-        final var modifiedAccounts = writableStore.modifiedAccountsInState();
+        // If there is a FROM_ACCOUNT_ or _TO_ACCOUNT stake change scenario, the set of modified
+        // accounts returned by writableStore.modifiedAccountsInState(); so we only iterate through
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java:105]: > the set of modified accounts returned by writableStore.modifiedAccountsInState();

I expected something to complete this statement, did I misread?

**Review by mhess-swl — APPROVED** [review]

**Review by kimbor — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]
LGTM

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java]: Ah fixed, oops.

**Review by iwsimon — DISMISSED** [review]

**Review by mhess-swl — DISMISSED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**